### PR TITLE
Add nats-server binary

### PR DESCRIPTION
**Pull request by Shikachuu**
## Description
Add nats-server binary.

## Motivation and Context

- [X] I have raised an issue to propose this change, which has been given a label of `design/approved` by a maintainer ([required](https://github.com/alexellis/arkade/blob/master/CONTRIBUTING.md)) #678

## How Has This Been Tested?
Covered with unit tests and tested with e2e test as well.

## Are you a GitHub Sponsor yet (Yes/No?)

- [ ] Yes
- [X] No

## Types of changes

- [ ] Bug fix (non-breaking change which fixes an issue)
- [X] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)

## Documentation

- [X] I have updated the list of tools in README.md if (required) with `./arkade get -o markdown`
- [ ] I have updated the list of apps in README.md if (required) with `./arkade install --help`

## Checklist:

- [X] My code follows the code style of this project.
- [X] My change requires a change to the documentation.
- [X] I have updated the documentation accordingly.
- [X] I've read the [CONTRIBUTION](https://github.com/alexellis/arkade/blob/master/CONTRIBUTING.md) guide
- [X] I have signed-off my commits with `git commit -s`
- [X] I have added tests to cover my changes.
- [X] All new and existing tests passed.

- [X] I have tested this on arm, or have added code to prevent deployment

### CODE REVIEW

**Comment by alexellis** [comment]
@Shikachuu thanks for the PR. Can you show the output from the "dagger tests" we have been using?

File: pkg/get/get_test.go [hunk 1]
```diff
@@ -3593,3 +3593,68 @@ func Test_DownloadHey(t *testing.T) {
...
+			url:     `https://github.com/nats-io/nats-server/releases/download/v2.7.4/nats-server-v2.7.4-linux-arm7.zip`,
+		},
+		{
+			os:      "ming",
+			arch:    arch64bit,
+			version: version,
+			url:     `https://github.com/nats-io/nats-server/releases/download/v2.7.4/nats-server-v2.7.4-windows-amd64.zip`,
+		},
+		{
+			os:      "ming",
```
Comment (alexellis) [inline pkg/get/get_test.go:3642]: I keep having to repeat this, but no arm for Windows for the time being. Thanks.

**Review by alexellis — CHANGES_REQUESTED** [review]
All good, but please remove the windows / ARM64

**Comment by Shikachuu** [comment]
> All good, but please remove the windows / ARM64

Done! :heavy_check_mark:

**Pull request merged**